tests/tasks/secrets/test_vault_secret.py

```diff
@@ -0,0 +1,184 @@
+import os
+import hvac
+import pytest
+import cloudpickle
+from typing import Any
+from unittest.mock import MagicMock
+
+# prefect imports
+from prefect import prefect, Task, Parameter
+from prefect.engine.results import SecretResult
+from prefect.utilities.tasks import defaults_from_attrs
+from prefect.exceptions import ClientError
+
+# local imports
+from prefect.tasks.secrets.vault_secret import VaultSecret
+
+
+def test_create_vault_var():
+    """
+    Verify class instance creation
+    """
+    vs = VaultSecret("fake-secret")
+    assert vs.name == "fake-secret"
+    assert isinstance(vs, VaultSecret)
+    assert isinstance(vs.result, SecretResult)
+
+
+def test_secret_is_pickleable():
+    """
+    Verify secret class instance can be pickeled
+    """
+    vs = VaultSecret("fake-pickeled-vault-secret")
+    new = cloudpickle.loads(cloudpickle.dumps(vs))
+    assert new.name == "fake-pickeled-vault-secret"
+    assert isinstance(new.result, SecretResult)
+
+
+# --- Test Harness Task ---
+class VaultSecretTestTask(Task):
+    """
+    Test harness task for testing VaultSecrets
+    Provides example of using VaultSecrets from a task
+
+    Returns a vault secret value from a vault secret key
+    """
+
+    def __init__(self, secret_key: str = None, **kwargs: Any):
+        self.secret_key = secret_key
+        kwargs.setdefault("name", secret_key)
+        super().__init__(**kwargs)
+
+    @defaults_from_attrs("secret_key")
+    def run(self, secret_key: str = None, vault_credentials_secret: str = None):
+        # self.logger.info("--- starting test task ---")
+        if vault_credentials_secret is None:
+            fake_secret = VaultSecret(self.secret_key)
+        else:
+            fake_secret = VaultSecret(
+                self.secret_key, vault_credentials_secret=vault_credentials_secret
+            )
+        fake_secret_val = fake_secret.run()
+        # self.logger.info("--- leaving test task ---")
+        return fake_secret_val
+
+
+# --- end VaultSecretTestTask ---
+
+
+def test_local_vault_secret():
+    """
+    Setting to use local secrets short-circuits vault lookup and the secret is
+    ready directly from the local secrets context.  Doesn't require a flow context.
+    Note that creating the explicit context seems to override the use_local_secrets
+    settings on the configuration.
+    """
+    fake_vault_secrets = {
+        "fake-local-secret": {"fake-local-secret-key": "fake-local-secret-value"}
+    }
+    vs = VaultSecret("fake-local-secret")
+    with prefect.context(secrets=fake_vault_secrets):
+        val = vs.run()
+        assert val == {"fake-local-secret-key": "fake-local-secret-value"}
+
+
+def test_vault_addr_env_var_missing(server_api):
+    prefect.context.config["cloud"]["use_local_secrets"] = False
+    os.unsetenv("vault_addr")
+    os.unsetenv("VAULT_ADDR")
```

Comment:
I believe you should use `monkeypatch.setenv` here instead of modifying the os directly

Reply:
`os.setenv()` and `os.unsetenv()`  calls replaced with monkey patch `setenv()` and `delenv()` calls.